src/main/java/org/janelia/saalfeldlab/paintera/meshes/MeshManagerWithAssignmentForSegments.java

```diff
@@ -233,21 +229,13 @@ public void removeMesh(final Long id)
 			mesh.isEnabledProperty().set(false);
 			mesh.interrupt();
 			mesh.meshSettingsProperty().unbind();
+			mesh.meshSettingsProperty().set(null);
 		}
 	}
 
-	private void removeMeshes(final Map<Long, MeshGenerator<TLongHashSet>> toBeRemoved)
+	private void removeMeshes(final Collection<Long> toBeRemoved)
 	{
-		toBeRemoved.values().forEach(mesh -> mesh.isEnabledProperty().set(false));
-		toBeRemoved.values().forEach(MeshGenerator::interrupt);
-
-		neurons.entrySet().removeAll(toBeRemoved.entrySet());
-		final List<Node> existingGroups = neurons.values().stream().map(MeshGenerator::getRoot).collect(Collectors.toList());
-		root.getChildren().setAll(existingGroups);
-
-		// unbind() for each mesh here takes way too long for some reason. Do it on a separate thread to avoid app freezing.
-		// TODO this should probably not create new threads, revisit!!
-		new Thread(() -> toBeRemoved.values().forEach(mg -> mg.meshSettingsProperty().unbind())).start();
+		toBeRemoved.forEach(this::removeMesh);
```

Comment:
Could you please test it in the following scenario: `Ctrl+A` thousands of labels (such as in CREMI sample A), then deselect all but one label. If I recall correctly, this special case implementation for removing a set of meshes at once helped to significantly increase performance in this case, otherwise it was taking forever if removing one by one.
Sorry for the ad-hoc optimization, I should've put some note in a comment about that.

Reply:
I'll test it on my mahcine.

Reply:
I do not observe it on my laptop with sample A, but I will also try on my workstation. What scale level did you try it on on your machine?

Reply:
I think the scale level didn't matter so it used to be slow even at the lowest resolution. The reason was that calling `removeMesh()` for each mesh would remove the groups from the scene one by one which has quadratic complexity compared to removing them at once, and also unbinding seemed to be very slow (the separate thread was a workaround for that).
Maybe this problem is gone though since you changed how the unbinding is done -- this would be great :)

Reply:
Strangely, it is slow on my workstation (but not on my laptop). I reverted to something more similar to what we had in c6ea9b9c and inverted the call chain to remove code duplication (`removeMesh` now calls `removeMeshes`). I introduced a new single-threaded `ExecutorService` with daemon threads that takes care of unbinding instead of creating new threads each time. I also tried to do similar optimizations for `generateMesh` (binding on a separate thread, adding all groups at once instead of one by one), but that did not seem to have an impact.